pkg/cmd/server/api/types.go

```diff
@@ -381,6 +381,10 @@ type OAuthConfig struct {
 	SessionConfig *SessionConfig
 
 	TokenConfig TokenConfig
+
+	// LoginTemplateFile is a file on the asset server to use as a template for login page customization.
+	// If unspecified, the default login page is used.
+	LoginTemplateFile string
```

Comment:
there are probably other pages we'll want to templatize (token display page, grant approval page, etc). Let's do a `Templates` object containing only a `Login` field for now